selendroid-standalone/src/main/java/io/selendroid/standalone/server/SelendroidStandaloneServer.java

```diff
@@ -68,11 +66,41 @@ protected SelendroidStandaloneDriver initializeSelendroidServer() throws Android
   public void start() {
     webServer.start();
     if (config.isGrid()) {
-      try {
-        new SelfRegisteringRemote(config, driver).performRegistration();
-      } catch (Exception e) {
-        log.log(Level.SEVERE, "Error registering selendroid into grid hub.", e);
+
+      class GridRegistrationTask extends TimerTask {
+        private final SelendroidStandaloneDriver driver;
+        private final SelendroidConfiguration configuration;
+        private final Logger log = Logger.getLogger(GridRegistrationTask.class.getName());
+        private boolean firstTime = true;
+
+        GridRegistrationTask(SelendroidStandaloneDriver driver, SelendroidConfiguration configuration) {
+          this.driver = driver;                // passing reference to driver and config
+          this.configuration = configuration;  // as, let's say, config might change during runtime... (device quantity)
+        }
+
+        @Override
+        public void run() {
+          try {
+            final SelfRegisteringRemote remote = new SelfRegisteringRemote(configuration, driver);
+
+            if (firstTime) {
```

Comment:
no need for this if, you can always set `firstTime` to `false`. The checking happens in `performRegistrationIfNotRegistered()`.